File: CoreAngularDemo.API/CoreAngularDemo.DAL/Models/Entities/Transition.cs
using System;
using CoreAngularDemo.DAL.Models.Entities.Abstractions;

namespace CoreAngularDemo.DAL.Models.Entities
{
    public class CoreAngularDemoion : IAuditableEntity, IBaseEntity
    {
        public int Id { get; set; }
        public int FromStateId { get; set; }
        public int ToStateId { get; set; }
        public int ActionTypeId { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public string CreatedById { get; set; }
        public string UpdatedById { get; set; }
        public bool IsFixed { get; set; }

        public virtual ActionType ActionType { get; set; }
        public virtual User Create { get; set; }
        public virtual State FromState { get; set; }
        public virtual User Mod { get; set; }
        public virtual State ToState { get; set; }
    }
}

File: CoreAngularDemo.Client/src/app/modules/shared/models/transition.ts
import { TEntity } from 'src/app/modules/core/models/entity/entity';
import { State } from './state';
import { ActionType } from './action-type';

export class CoreAngularDemoion extends TEntity<CoreAngularDemoion> {
  fromState: State;
  toState: State;
  actionType: ActionType;
  isFixed: boolean;

  constructor(CoreAngularDemoion: Partial<CoreAngularDemoion>) {
    super(CoreAngularDemoion);
    this.fromState = new State(this.fromState);
    this.toState = new State(this.toState);
    this.actionType = new ActionType(this.actionType);
    this.isFixed = !!this.isFixed;
  }
}
